resources/asciidoctor/spec/elastic_compat_tree_processor_spec.rb

```diff
@@ -72,51 +74,82 @@
     expect(actual).to eq(expected.strip)
   end
 
-  [
-    %w[CONSOLE console],
-    %w[AUTOSENSE sense],
-    %w[KIBANA kibana],
-    %w[SENSE:path/to/snippet.sense sense],
-  ].each do |command, lang|
-    it "transforms legacy // #{command} commands into the #{lang} language" do
-      actual = convert <<~ASCIIDOC
-        == Example
-        [source,js]
-        ----
-        GET /
-        ----
-        pass:[// #{command}]
-      ASCIIDOC
-      expected = <<~DOCBOOK
-        <chapter id="_example">
-        <title>Example</title>
-        <programlisting language="#{lang}" linenumbering="unnumbered">GET /</programlisting>
-        </chapter>
-      DOCBOOK
-      expect(actual).to eq(expected.strip)
-    end
-  end
+  shared_examples 'snippet language' do |override, lang|
+    name = override ? " the #{override} lang override" : 'out a lang override'
+    context "for a snippet with#{name}" do
+      let(:snippet) do
+        snippet = <<~ASCIIDOC
+          [source,js]
+          ----
+          GET / <1>
+          ----
+        ASCIIDOC
+        snippet += "lang_override::[#{override}]" if override
+        snippet
```

Comment:
Have you seen the `#tap` method in Ruby? It's a style thing, but you could use it here. I like using it so I never accidentally forget to return the object (in this case, `snippet`) from the let block.

Reply:
`#tap` seems pretty complex to use here because the string is frozen. Though I bet there are other places i can use it....

Reply:
ah, then it wouldn't work if the String is frozen..